aten/src/ATen/core/type.cpp

```diff
@@ -753,6 +764,265 @@ TupleTypePtr TupleType::createNamed(
       field_types, qualName, schema)); // NOLINT(modernize-make-shared)
 }
 
+bool NoneType::isSubtypeOfExt(const TypePtr& rhs, std::ostream *why_not) const {
+  if (rhs->kind() == OptionalType::Kind) {
+    return true;
+  }
+  if (auto rhs_ = rhs->cast<UnionType>()) {
```

Comment:
do wee need to add this logic? won't it naturally be covered by `Type::isSubtypeOfExt` ?